frontend/src/components/sign/components/sign-up-form/sign-up-form.tsx

```diff
@@ -1,29 +1,52 @@
-import { useState } from 'react';
-import { Link } from 'react-router-dom';
+import React from 'react';
+import { FieldValues } from 'react-hook-form';
+import { Link, useNavigate } from 'react-router-dom';
 
+import { SignUpRequestData } from '@autoline/shared/common/types/types';
+import { signUpSchema as baseSchema } from '@autoline/shared/validation-schemas';
 import { AppRoute } from '@common/enums/app/app-route.enum';
 import { ButtonFill } from '@components/common/button-fill/button-fill';
 import { ButtonOutline } from '@components/common/button-outline/button-outline';
 import { InputField } from '@components/common/input-field/input-field';
-import { SelectField } from '@components/common/select-field/select-field';
-import { MenuItem, SelectChangeEvent } from '@mui/material';
+import { useAppForm } from '@hooks/hooks';
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
 import Divider from '@mui/material/Divider';
+import { ErrorType } from '@store/queries';
+import { useSignUpMutation } from '@store/queries/auth';
+import * as Yup from 'yup';
 
 import styles from './styles.module.scss';
 
+const signUpSchema = baseSchema.shape({
+  repeatPassword: Yup.string()
+    .required('Please, repeat the password')
+    .oneOf([Yup.ref('password'), null], 'Passwords must match'),
+});
+
 export const SignUpForm = (): React.ReactElement => {
-  const [isLoading, setIsLoading] = useState(false);
-  const [location, setLocation] = useState('');
-  const handleSubmit = (): void => {
-    setIsLoading(true);
-  };
+  const { control, errors, handleSubmit } = useAppForm<SignUpRequestData>({
+    defaultValues: {},
+    validationSchema: signUpSchema,
+  });
+
+  const navigate = useNavigate();
+  const [fetchSignUp, { isLoading, isSuccess, isError, error: fetchError }] =
+    useSignUpMutation();
```

Comment:
`fetchSignUp` isn't semantically correct :slightly_smiling_face: 
Because `fetch` is the term for receiving smth from the GET request. But here we use the POST request. So the `signUp` would be better